spyder/plugins/maininterpreter.py

```diff
@@ -153,6 +158,28 @@ def python_executable_switched(self, custom):
             if custom:
                 self.warn_python_compatibility(cust_pyexec)
 
+    def is_python_interpreter(self, pyexec):
+        if not osp.isfile(pyexec):
+            return False
+        args = ["-c", "print('valid_python_interpreter')"]

```

Comment:
You can also run in a terminal:

    ipython -c 'print("foo")'

so this is not enough for a test.